### PR TITLE
Add minimal documentation

### PR DESCRIPTION
**Pull request by Parzival-05**

### CODE REVIEW

**Review by emnigma — CHANGES_REQUESTED** [review]
Check out this github repo, it has nice structure: https://github.com/ahmedfgad/GeneticAlgorithmPython

File: README.md [hunk 1]
```diff
@@ -37,8 +39,35 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
-### Run training
-...
+Usage pipeline:
+
+- Collect a dataset: set of functions or methods. You can use `./workflow/dataset_for_tests_java.yml` as a template.
+- Configuration setup (specifying server and training parameters). You can use `./workflow/config_for_tests.yml` as a template.
+  - Below we will discuss how you can fully integrate your own symbolic machine
+- Launching the server manager (`launch_servers`)
+- Run the training process (`run_training`) to get model.
+- Use ONNX to convert your model to ONNX format.
```
[the remark this answers is not in the document]
Comment (Parzival-05) [reply README.md]: Fixed: https://github.com/PySymGym/PySymGym/pull/151/commits/91d5a651cf34d4d283aafbaf59470719eda3a09c

File: README.md [hunk 2]
```diff
@@ -37,8 +39,35 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
+- Use your ONNX model to run symbolic execution with the model.
+
+### Integrate a new symbolic machine
```
[the remark this answers is not in the document]
Comment (Parzival-05) [reply README.md:54]: Fixed: https://github.com/PySymGym/PySymGym/commit/91d5a651cf34d4d283aafbaf59470719eda3a09c

**Comment by Anya497** [comment]
Could you rebase and merge this pull request into [readme_update](https://github.com/PySymGym/PySymGym/tree/readme_update) branch please? We already have there some important changes.

File: README.md [hunk 3]
```diff
@@ -37,8 +39,37 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
+Usage pipeline:
+
+- Collect a dataset: set of functions or methods. You can use [`./workflow/dataset_for_tests_java.yml`](./workflow/dataset_for_tests_java.yml) as a template.
+- Create configuration (specifying server and training parameters). You can use [`./workflow/config_for_tests.yml`](./workflow/config_for_tests.yml) as a template.
+  - Below we will discuss how you can fully integrate your own symbolic machine
+- Launch the server manager
+  - `poetry --ansi run poe launch_servers --config path/to/config.yml`
+- Run the training process to get PyTorch model.
+  - `poetry --ansi run poe run_training --config path/to/config.yml`
+- Use ONNX to convert your PyTorch model to ONNX format [(see below how to)](#onnx-conversion).
```
Comment (emnigma) [inline README.md:51]: This is still not quite right, we do not use ONNX to convert pytorch model to `.onnx`. If you want to be specific, then the conversion happens with `torch.onnx` module of `pytorch` lib:
image

However, I advise you to not specify this. Instead we can write something like this:
```
Use `onyx.py`(link) command line tool to convert ...
```

**Review by emnigma — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -53,11 +82,12 @@ onyx.py --sample-gamestate <game_state0.json> \
     [optional] --verify-on <game_state1.json> <game_state2.json> <game_state3.json> ...
 ```
 
-model_kwargs yaml file, *verification* game_states and *sample* game_state (use any) can be found in [resources/onnx](resources/onnx/) dir
+model*kwargs yaml file, \_verification* game*states and \_sample* game_state (use any) can be found in [resources/onnx](resources/onnx/) dir
```
[the remark this answers is not in the document]
Comment (Parzival-05) [reply README.md:87]: Oh, my auto formatter set me up. I will fix this before merging.

**Comment by Parzival-05** [comment]
> Could you rebase and merge this pull request into [readme_update](https://github.com/PySymGym/PySymGym/tree/readme_update) branch please? We already have there some important changes.

done

File: README.md (hunk 3 above)
Comment (Parzival-05) [reply README.md]: done

File: README.md [hunk 5]
```diff
@@ -51,27 +53,64 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
+
+- Build game server
+- Build game maps
+- Create Json with maps description
+  - You can use `./workflow/dataset_for_tests_java.json` as a template
+- Generate initial data
+- Convert initial data to dataset for training
+- Run training process
+- Use [`onyx.py`](#onnx-conversion) command line tool convert your PyTorch model to ONNX format.
+- Use your ONNX model to run symbolic execution.
```
Comment (emnigma) [inline README.md:71]:
```suggestion
- Use your ONNX model to guide symbolic execution.
```

File: README.md [hunk 6]
```diff
@@ -51,27 +53,64 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
+- See [play_game](AIAgent/ml/game/play_game.py) to implement a server part with support for the websocket protocol
+- See [messages](AIAgent/connection/game_server_conn/messages.py) to provide serialization-deserialization of data according to the established protocol
+- Implement methods for:
+
+  - Symbolic execution in training mode
+  - Running with a trained model
+
+_Integration examples:_
+
+- [VSharp](GameServers/VSharp/) (.NET symbolic machine) and [its maps](maps/DotNet/)
```
Comment (emnigma) [inline README.md:112]: links for VSharp and usvm don't work

File: README.md [hunk 7]
```diff
@@ -51,27 +53,64 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
+
+![Illustration of game process](resources/game_process_illustration.svg)
+
+To integrate a new symbolic machine, it is necessary to:
+
+- See [play_game](AIAgent/ml/game/play_game.py) to implement a server part with support for the websocket protocol
+- See [messages](AIAgent/connection/game_server_conn/messages.py) to provide serialization-deserialization of data according to the established protocol
+- Implement methods for:
+
+  - Symbolic execution in training mode
```
Comment (emnigma) [inline README.md:107]:
```suggestion
  - Symbolic execution in training mode (example: `VSharp/VSharp.ML.GameServer.Runner`)
```

File: README.md [hunk 8]
```diff
@@ -51,27 +53,64 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
+  - Running with a trained model
```
Comment (emnigma) [inline README.md:108]:
```suggestion
  - Running with a trained model (example: `VSharp.Explorer/AISearcher.fs`)
```

**Review by emnigma — APPROVED** [review]
a few fix minor things and its good to go

File: README.md (hunk 6 above)
Comment (Parzival-05) [reply README.md]: fixed

**Comment by Parzival-05** [comment]
@Anya497 @gsvgit

File: README.md [hunk 9]
```diff
@@ -51,27 +50,68 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
-- Build game server
-- Build game maps
-- Create Json with maps description 
-- Generate initial data
-- Convert initial data to dataset for training
-- Run training process
+- Create configuration (specifying server and training parameters). You can use [`./workflow/config_for_tests.yml`](./workflow/config_for_tests.yml) as a template.
+  - Below we will discuss how you can fully integrate your own symbolic machine
+- Launch the server manager
+  - `poetry run python3 launch_servers --config path/to/config.yml`
```
Comment (Anya497) [inline README.md:90]:
```suggestion
  - `poetry run python3 launch_servers.py --config path/to/config.yml`
```

File: README.md [hunk 10]
```diff
@@ -51,27 +50,68 @@ Then follow installation instructions provided on [torch](https://pytorch.org/ge
...
+- Run MLFlow server
+  - `poetry run mlflow server -h 127.0.0.1 -p 8080 --serve-artifacts`
+- Run the training process to get PyTorch model.
+  - `poetry run python3 run_training --config path/to/config.yml`
```
Comment (Anya497) [inline README.md:94]:
```suggestion
  - `poetry run python3 run_training.py --config path/to/config.yml`
```

**Review by Anya497 — CHANGES_REQUESTED** [review]

**Comment by gsvgit** [comment]
Can sequence diagram from the paper be reused?

**Comment by Parzival-05** [comment]
> Can sequence diagram from the paper be reused?

Where do you think this diagram would look more natural in the readme? You can insert it by yourself

**Comment by Parzival-05** [comment]
@gsvgit

**Review by gsvgit — APPROVED** [review]
